lte/gateway/c/core/oai/test/ngap/test_ngap_state_converter.cpp

```diff
@@ -78,9 +99,42 @@ TEST_F(NGAPStateConverterTest, NgapStateConversionSuccess) {
   EXPECT_EQ(gnbd->sctp_assoc_id, gnbd_final->sctp_assoc_id);
   EXPECT_EQ(gnbd->gnb_id, gnbd_final->gnb_id);
   EXPECT_EQ(gnbd->ng_state, gnbd_final->ng_state);
+  EXPECT_EQ(gnbd->instreams, gnbd_final->instreams);
+  EXPECT_EQ(gnbd->outstreams, gnbd_final->outstreams);
+  EXPECT_EQ(gnbd->next_sctp_stream, gnbd_final->next_sctp_stream);
+  EXPECT_EQ(gnbd->nb_ue_associated, gnbd_final->nb_ue_associated);
+  EXPECT_EQ(gnbd->default_paging_drx, gnbd_final->default_paging_drx);
+  EXPECT_EQ(
+      memcmp(gnbd->gnb_name, gnbd_final->gnb_name, sizeof(gnbd->gnb_name)), 0);
+
+  EXPECT_EQ(
+      hashtable_uint64_ts_get(&gnbd->ue_id_coll, (const hash_key_t) 1, &data),
+      HASH_TABLE_OK);
+  EXPECT_EQ(data, 17);
+  data = 0;
+  EXPECT_EQ(
+      hashtable_uint64_ts_get(
+          &gnbd_final->ue_id_coll, (const hash_key_t) 2, &data),
+      HASH_TABLE_OK);
+  EXPECT_EQ(data, 25);
+  EXPECT_EQ(
+      hashtable_ts_get(
+          &init_state->amfid2associd, (const hash_key_t) 1,
+          reinterpret_cast<void**>(&associd1)),
+      HASH_TABLE_OK);
+  EXPECT_EQ(
+      hashtable_ts_get(
+          &final_state->amfid2associd, (const hash_key_t) 1,
```

Comment:
Why is there a lookup for same amf id twice? Shouldn't it be 2 here?

```suggestion
          &final_state->amfid2associd, (const hash_key_t) 2,
```

Reply:
Thanks for the review @ssanadhya.
Here we are verifying that whether entries for init state and final state amfid2associd are present. 
On line 122 we are verifying for init state and on line 127 we are verifying for the final state amfid2associd.

Reply:
Thanks for the clarification @aniket021997 ! I missed that part. Looks good.